File: smart_farm/src/api/cameraImage.ts
import type { AxiosResponse } from 'axios'
import { http } from '@/utils/request'
import type { PageInfo } from '@/api/user'

export interface CameraImage {
  id?: number
  cameraId: number
  cameraName?: string
  plotId: number
  plotName?: string
  plotCode?: string
  userId?: number
  username?: string
  nickname?: string
  imageUrl: string
  imageSize?: number
  captureTime?: string
  remark?: string
  createTime?: string
  updateTime?: string
}

export interface CameraImageListParams {
  cameraId?: number
  plotId?: number
  cameraName?: string
  plotName?: string
  startDate?: string
  endDate?: string
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

export const listCameraImages = (params: CameraImageListParams) => {
  return http<ApiResult<PageInfo<CameraImage>>>({
    url: '/camera-image/list',
    method: 'get',
    params: params as unknown as Record<string, unknown>,
    preserveParamDateTime: true,
  })
}

export const deleteCameraImage = (id: number) => {
  return http<ApiResult<void>>({ url: `/camera-image/${id}`, method: 'delete' })
}

export const batchDeleteCameraImages = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/camera-image/batch', method: 'delete', data: ids })
}

export const downloadCameraImage = (id: number) => {
  return http<AxiosResponse<Blob>>({
    url: `/camera-image/${id}/download`,
    method: 'get',
    responseType: 'blob',
  })
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/CameraImage.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CameraImage {

    private Long id;

    private Long cameraId;

    private String cameraName;

    private Long batchId;

    private Long plotId;

    private String plotName;

    private String plotCode;

    private Long userId;

    private String username;

    private String nickname;

    private String imageUrl;

    private String thumbnailUrl;

    private Long imageSize;

    private Integer captureType;

    private LocalDateTime captureTime;

    private Integer aiChecked;

    private Integer status;

    private String remark;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}
